test/string-utils.cjs

```diff
@@ -33,4 +40,102 @@ describe('string-utils', function () {
       strictEqual(looksLikeNumber('apple'), false)
     })
   })
+
+  describe('parseAnsiCQuotedString', () => {
+    async function check (ansiString, encoding = 'utf8') {
+      const result = parseAnsiCQuotedString(ansiString)
+
+      const resultBuffer = Buffer.from(result, encoding)
+      const firstNull = resultBuffer.indexOf(0)
+      const resultUpToFirstNull = firstNull === -1 ? resultBuffer : resultBuffer.slice(0, firstNull)
+
+      const { stdout } = await exec(
+        'printf "%s" ' + ansiString,
+        { shell: '/bin/bash', encoding: 'buffer' }
+      )
+
+      if (!resultUpToFirstNull.equals(stdout)) {
+        console.log(JSON.stringify(ansiString), ansiString[4], ansiString.codePointAt(4))
+        console.log(resultUpToFirstNull, 'should be', stdout)
+        if (!resultUpToFirstNull.equals(resultBuffer)) {
+          console.log('(', resultUpToFirstNull, 'is actually', resultBuffer, ')')
+        }
+        console.log()
+      }
+
+      resultUpToFirstNull.should.deep.equal(stdout)
+    }
+
+    async function checkAll (prefix, value, maxLength, encoding = 'utf8') {
+      await check("$'\\" + prefix + value + "'", encoding)
+      while (value.length < maxLength) {
+        value = '0' + value
+        await check("$'\\" + prefix + value + "'", encoding)
+        if (value.toLowerCase() !== value) {
+          await check("$'\\" + prefix + value.toLowerCase() + "'", encoding)
+        }
+      }
+    }
+
+    // Skip exhaustive testing because it takes a long time
+    xit('parses all octal codes in ANSI-C quoted strings like bash', async () => {
```

Comment:
How do you run these tests? Perhaps you could add an additional `test-exhaustive` script to `package.json`, and we could run them just when we merge to the main branch.